JustSaying.UnitTests/XBehaviourTest.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using AutoFixture;
+
+namespace JustSaying.UnitTests
+{
+    public abstract class XBehaviourTest<TSystemUnderTest>
+    {
+        private bool _recordExceptions;
+
+        protected XBehaviourTest()
+        {
+            Execute();
+        }
+
+        protected TSystemUnderTest SystemUnderTest { get; private set; }
+        protected Exception ThrownException { get; private set; }
+ 
+        protected virtual TSystemUnderTest CreateSystemUnderTest()
+        {
+            var fixture = new Fixture();
+            return fixture.Create<TSystemUnderTest>();
+        }
+
+        protected void Execute()
+        {
+            Given();
+
+            try
+            {
+                SystemUnderTest = CreateSystemUnderTest();
+                When();
+            }
+            catch (Exception ex)
```

Comment:
Could simplify this (I think) to:

```csharp
catch (Exception ex) when (_recordExceptions)
{
    ThrownException = ex;
}
```

Reply:
Ah the infamous `when` keyword, as avoided in this PR!